jobs/build/build-sync/build.groovy

```diff
@@ -127,8 +127,16 @@ def buildSyncApplyImageStreams() {
 
         // Get the current IS and save it to disk. We may need this for debugging.
         def currentIS = getImageStream(theStream, namespace)
-        writeJSON(file: "pre-apply-${namespace}-${theStream}.json", json: currentIS)
-        artifacts.addAll(["pre-apply-${namespace}-${theStream}.json"])
+        def currentISfilename = "pre-apply-${namespace}-${theStream}.json"
+        writeJSON(file: currentISfilename, json: currentIS)
+        artifacts.addAll([currentISfilename])
+
+        // Make sure there's still an update to IS
+        def diffStatus = buildlib.oc("--kubeconfig ${buildlib.ciKubeconfig} diff --filename=${isFile}", [capture: true, returnStatus: true])
```

Comment:
I am not sure this will do the right thing. In kubernetes, admission controllers and other operators may add additional fields (like `annotations`) to the resource. This may always produce a diff. But I could be wrong.

Reply:
`oc apply` is idempotent. If we don't want to it fail in case of no change, why not deleting the code that checks the resource version?

Reply:
I agree simpler would be to just remove the verification. In the past there were cases when imagestream did not update - oc apply failed for some reason - I don't know if they still happen? But I would say since this is better than just plain failing - it does skip when there is not change to make - see the linked test run. Maybe it still produces a diff in some cases - if it does let's come back and fix that?